clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java

```diff
@@ -242,6 +377,60 @@ public AttributeList setAttributes(AttributeList list) {
             throw new UnsupportedOperationException("Set not allowed.");
         }
 
-    }
+        /**
+         * Add some new metrics to this mbean, then unregister and re-register the mbean.
+         */
+        void addMetrics(Collection<KafkaMetric> metrics) {
+            checkLocked();
+            if (!beanMetrics.isEmpty()) {
+                unregister();
+            }
+            for (KafkaMetric metric : metrics) {
+                beanMetrics.put(metric.metricName().name(), metric);
+            }
+            register();
+        }
+
+        void removeMetrics(Collection<KafkaMetric> metrics) {
+            checkLocked();
+            if (!beanMetrics.isEmpty()) {
+                unregister();
+            }
+            for (KafkaMetric metric : metrics) {
+                beanMetrics.remove(metric.metricName().name(), metric);
+            }
+            if (!beanMetrics.isEmpty()) {
+                register();
+            }
+        }
+
+        void register() {
+            checkLocked();
+            try {
+                owner.mbeanServer.registerMBean(this, objectName);
+            } catch (Exception e) {
+                throw new RuntimeException("Failed to reregister bean " + mbeanName, e);
```

Comment:
Was there any reason to change these from `KafkaException`?

Reply:
No good reason, I just forgot that they were using `KafkaException`.  I will change it back.